aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -147,8 +173,10 @@ module aptos_framework::code {
             let j = 0;
             while (j < vector::length(new_modules)) {
                 let name = vector::borrow(new_modules, j);
-                assert!(&old_mod.name != name, error::already_exists(EMODULE_NAME_CLASH))
-            }
+                assert!(&old_mod.name != name, error::already_exists(EMODULE_NAME_CLASH));
+                j = j + 1;
+            };
+            i = i + 1;
```

Comment:
I find it interesting this code had 2 for loops without adjusting the counters prior.